test/include/test.hpp

```diff
@@ -654,20 +737,38 @@ struct driver
         {
             std::unordered_map<std::string, test_case> m(get_test_cases().begin(),
                                                          get_test_cases().end());
+
             for(auto&& iname : cases)
             {
-                for(auto&& name : get_case_names(iname))
+                std::vector<std::pair<std::string, test_case>> found_cases;
+                for(auto&& pattern : get_case_names(iname))
                 {
-                    auto f = m.find(name);
+                    auto f = m.find(pattern);
                     if(f == m.end())
                     {
-                        out() << color::fg_red << "[  ERROR   ] Test case '" << name
-                              << "' not found." << color::reset << std::endl;
-                        failed.push_back(name);
+                        std::copy_if(get_test_cases().begin(),
```

Comment:
Since you said I wasn't focused on the solution (!!)
Here is a quick replacement for all your custom 'glob' code, with what the standard library offers. About **half** the size of your code -- not including comments. And it can always be tweaked depending on your requirements, which I see are using only '*' and '?'. I did not run the test.cpp, but that should likely be fine.

```
                    if(f == m.end())
                    {
                        std::string esc_pattern;
                        const auto meta = {
                            '.', '^', '$', '*', '+', '?', '(', ')', '[', ']', '{', '}', '\\', '|'};

                        for(auto c : pattern)
                        {
                            if(c == '*' || c == '?')
                                esc_pattern.append(1, '.');
                            else if(std::find(meta.begin(), meta.end(), c) != meta.end())
                                esc_pattern.append("\\");
                            esc_pattern.append(1, c);
                        }

                        std::regex rex_pattern(esc_pattern);
                        for(const auto& test_name : get_test_cases())
                            if(std::regex_match(test_name.first, rex_pattern))
                                found_cases.push_back(test_name);
                    }
```

Reply:
I dont think this handles the translation correctly. I already found one test case where it fails. It can be easily fixed, but my concern is that there is more cases that may not be handled correctly either. A compete translation that is used in libraries like p-ranav/glob [here](https://github.com/p-ranav/glob/blob/master/source/glob.cpp#L21) are much bigger, so there could defintately be some cases we are still missing. 

Furthermore, fixing those issue makes the function almost as big as the `glob_match` so its not really that much simpler.

Reply:
@sean-parent, thanks to your hint at _the complexity of transforming a glob pattern to a regex pattern_ it made me do some reading. My simple transform logic presented above doesn't properly handle (though easily fixable) an expected 'failure' case very well, as regex engine will become very slow: For example: to match "aaaaaaaaaaaaaaaaaaaaa" with a pattern "a**z" should fail, but with a N**2 slowdown, and a***z with N**3 slowdown. N being the length of the input string..  

Therefore, we should add this (or similar) negative test to our test cases. Thanks.